app/assets/javascripts/discourse/app/controllers/tag-show.js

```diff
@@ -1,226 +1,198 @@
-    if (loading) {
-      return;
-    }
-
-    if (listTopicsLength === 0) {
-      return I18n.t(`tagging.topics.none.${navMode}`, {
-        tag: this.get("tag.id"),
-      });
-    } else {
-      return I18n.t(`topics.bottom.tag`, {
-        tag: this.get("tag.id"),
-      });
-    }
-  },
-
-  @discourseComputed("list.filter", "list.topics.length")
-  showDismissRead(filter, topicsLength) {
-    return this._isFilterPage(filter, "unread") && topicsLength > 0;
-  },
-
-  @discourseComputed("list.filter", "list.topics.length")
-  showResetNew(filter, topicsLength) {
-    return this._isFilterPage(filter, "new") && topicsLength > 0;
-  },
-
-  actions: {
-    resetNew() {
-      const tracked =
-        (this.router.currentRoute.queryParams["f"] ||
-          this.router.currentRoute.queryParams["filter"]) === "tracked";
-
-      let topicIds = this.selected
-        ? this.selected.map((topic) => topic.id)
-        : null;
-
-      Topic.resetNew(this.category, !this.noSubcategories, {
-        tracked,
-        tag: this.tag,
-        topicIds,
-      }).then(() =>
-        this.send(
-          "refresh",
-          tracked ? { skipResettingParams: ["filter", "f"] } : {}
-        )
+export default DiscoverySortableController.extend(
+  BulkTopicSelection,
+  FilterModeMixin,
+  {
+    application: controller(),
+
+    tag: null,
+    additionalTags: null,
+    list: null,
+    canAdminTag: readOnly("currentUser.staff"),
+    navMode: "latest",
+    loading: false,
+    canCreateTopic: false,
+    showInfo: false,
+    top: endWith("list.filter", "top"),
+
+    @discourseComputed(
+      "canCreateTopic",
+      "category",
+      "canCreateTopicOnCategory",
+      "tag",
+      "canCreateTopicOnTag"
+    )
+    createTopicDisabled(
+      canCreateTopic,
+      category,
+      canCreateTopicOnCategory,
+      tag,
+      canCreateTopicOnTag
+    ) {
+      return (
+        !canCreateTopic ||
+        (category && !canCreateTopicOnCategory) ||
+        (tag && !canCreateTopicOnTag)
       );
     },
 
-    showInserted() {
-      const tracker = this.topicTrackingState;
-      this.list.loadBefore(tracker.get("newIncoming"), true);
-      tracker.resetTracking();
-      return false;
+    @discourseComputed("category", "tag.id", "filterType", "noSubcategories")
+    navItems(category, tagId, filterType, noSubcategories) {
+      return NavItem.buildList(category, {
+        tagId,
+        filterType,
+        noSubcategories,
+        siteSettings: this.siteSettings,
+      });
     },
 
-    changeSort(order) {
-      if (order === this.order) {
-        this.toggleProperty("ascending");
-      } else {
-        this.setProperties({ order, ascending: false });
-      }
-
-      let params = { order, ascending: this.ascending };
-      if (this.period) {
-        params.period = this.period;
-      }
-
-      this.transitionToRoute({
-        queryParams: params,
-      });
+    @discourseComputed("category")
```

Comment:
what is the purpose of this? Why do we recompute on category if we always return true ?

Reply:
Good question! I can't find anywhere that consumes this property, so I'll remove it completely